File: Flipping_Image/flip.js
/**
 * @param {number[][]} image - 2D matrix with bit image data
 * @returns {number[][]} returns the flip horizontal of matrix
 */
var flipAndInvertImage = function (image) {
	if (image.length === 0 || image[0].length === 0) {
		return [];
	}

	const flipImage = [];

	for (let i = 0; i < image.length; i++) {
		flipImage[i] = [];
		for (let j = image[0].length - 1; j >= 0; j--) {
			flipImage[i].push(image[i][j] ^ 1);
		}
	}

	return flipImage;
};

File: Flipping_Image/flip.py
from typing import List


class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        flip_image: List[List[int]] = []
        if len(image) == 0 or len(image[0]) == 0:
            return flip_image

        i = 0

        while i < len(image):
            flip_row = []

            j = len(image[0]) - 1

            while j >= 0:
                flip_row.append(image[i][j] ^ 1)

                j -= 1

            flip_image.append(flip_row)
            i += 1

        return flip_image
